### PR TITLE
[All habits] Refactor GET/habit/assign/{habitId}/allUserAndCustomList

### PR DESCRIPTION
**Pull request by sergoliarnik**
## Summary Of Issue :
Refactor GET/habit/assign/{habitId}/allUserAndCustomList endpoints, so it accepts habitAssignId instead habitId

**Issue Link**
https://github.com/ita-social-projects/GreenCity/issues/5522

## Summary Of Changes :
1) refactor getUserShoppingAndCustomShoppingLists controller method
2) add getUserShoppingListByHabitAssignId method to ShoppingListItemService
3) add findAllAvailableCustomShoppingListItemsByHabitAssignId method to CustomShoppingListItemService
5) refactor tests

## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [ ] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/main/java/greencity/controller/HabitAssignController.java [hunk 1]
```diff
@@ -150,27 +150,27 @@ public ResponseEntity<List<HabitAssignDto>> getCurrentUserHabitAssignsByIdAndAcq
...
-     * @param habitId {@link HabitVO} id.
-     * @param userVO  {@link UserVO} instance.
-     * @param locale  needed language code.
+     * @param habitAssignId {@link HabitAssignVO} id.
+     * @param userVO        {@link UserVO} instance.
+     * @param locale        needed language code.
      * @return User Shopping List and Custom Shopping List.
      */
-    @ApiOperation(value = "Get user shopping and user custom shopping lists")
+    @ApiOperation(value = "Get user shopping and custom shopping lists")
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitAssignController.java]: please add `by habitAssignId`

File: core/src/main/java/greencity/controller/HabitAssignController.java [hunk 2]
```diff
@@ -431,22 +431,21 @@ public ResponseEntity<HabitAssignDto> cancelHabitAssign(@PathVariable Long habit
     }
 
     /**
-     * Method delete assigned habit {@link HabitAssignVO} with inprogress status for
-     * current {@link UserVO}.
+     * Method delete habit assign {@link HabitAssignVO} for current {@link UserVO}.
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitAssignController.java]: please add `by habitAssignId`

File: core/src/main/java/greencity/controller/HabitAssignController.java [hunk 3]
```diff
@@ -431,22 +431,21 @@ public ResponseEntity<HabitAssignDto> cancelHabitAssign(@PathVariable Long habit
...
-     * @param userVO  - {@link UserVO} user.
+     * @param habitAssignId - id of {@link HabitAssignVO}.
+     * @param userVO        - {@link UserVO} user.
      */
-    @ApiOperation(value = "Delete assigned habit with inprogress status for current user.")
+    @ApiOperation(value = "Delete habit assign by habitAssignId for current user.")
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = HttpStatuses.OK),
         @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
         @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitAssignController.java:446]: can we have `400` response?
For example, if we use `String` as `habitAssignId`

File: core/src/main/java/greencity/controller/HabitAssignController.java [hunk 4]
```diff
@@ -150,27 +150,27 @@ public ResponseEntity<List<HabitAssignDto>> getCurrentUserHabitAssignsByIdAndAcq
...
+     * @param locale        needed language code.
      * @return User Shopping List and Custom Shopping List.
      */
-    @ApiOperation(value = "Get user shopping and user custom shopping lists")
+    @ApiOperation(value = "Get user shopping and custom shopping lists")
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = HttpStatuses.OK,
             response = UserShoppingAndCustomShoppingListsDto.class),
         @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
         @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/HabitAssignController.java]: cam we have `404` response?
For example, if there are no record in the database with `habitAssignId` or this `habitAssignId` does not belong to current user

File: service/src/main/java/greencity/service/CustomShoppingListItemServiceImpl.java [hunk 5]
```diff
@@ -230,6 +235,30 @@ public List<CustomShoppingListItemResponseDto> findAllAvailableCustomShoppingLis
...
+
+        Long habitId = habitAssign.get().getHabit().getId();
+
+        List<CustomShoppingListItem> customShoppingListItemList =
+            customShoppingListItemRepo.findAllAvailableCustomShoppingListItemsForUserId(userId, habitId);
+
+        List<CustomShoppingListItemResponseDto> responseDtoList = new ArrayList<>();
+
+        customShoppingListItemList
+            .forEach(item -> responseDtoList.add(modelMapper.map(item, CustomShoppingListItemResponseDto.class)));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/CustomShoppingListItemServiceImpl.java]: use collect to list

**Review by RomaMocherniuk — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Pull request merged**